lib/OktaAuth.ts

```diff
@@ -244,19 +245,17 @@ class OktaAuth implements SigninAPI, SignoutAPI {
     return isInteractionRequiredError(error);
   }
 
-  /**
-   * Alias method of signInWithCredentials
-   * 
-   * @todo This method is deprecated. Remove it in 5.0
-   */ 
-  signIn(opts) {
-    if (this.features.isLocalhost()) {
-      deprecate('This method has been deprecated, please use signInWithCredentials() instead.');
+  async signIn(opts: SigninOptions = { useInteractionCodeFlow: true }): Promise<AuthTransaction> {
+    const useInteractionCodeFlow = opts.useInteractionCodeFlow || this.options.useInteractionCodeFlow;
+    if (useInteractionCodeFlow) {
+      return authenticate(this, opts);
```

Comment:
looks like here is the only place `authenticate` gets exposed in the SDK. Do we also want it to be exposed from `idx` namespace to follow the same signature across all SDKs?

Reply:
Yes, I think so. `signIn` and `signInWithCredentials` here are "convenience" methods, they offer compatibility with Authn V1.  `authenticate` is the underlying method for IDX only.